### ISSUE TITLE
Frozen runpy When Trying to Run AutoGPT via Python 3.11.3 (Windows 10 cmd Virtual Env)

### ISSUE THREAD

**Issue by ZylerKade** [Significant-Gravitas/Auto-GPT#4109]
### ⚠️ Search for existing issues first ⚠️

- [X] I have searched the existing issues, and there is no existing issue for my problem

### Which Operating System are you using?

Windows

### Which version of Auto-GPT are you using?

Stable (branch)

### GPT-3 or GPT-4?

GPT-4

### Steps to reproduce 🕹

1. Latest versions of Python and PIP installed
2. Download Latest Stable AutoGPT version from GitHub (current v0.3.0)
3. Open CMD
4. Change Directory to AutoGPT Root - "cd C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0"
5. Create Virtual Environment - "python -m venv venv"
6. Activate Virtual Environment - "venv\Scripts\activate"
7. Install requirements - "pip install -r requirements.txt"
8. Attempt to run AutoGPT - "python -m autogpt"
9. News message loads, but then followed by "frozen runpy" error.
10. Trying to use/run with VS Code + Containers works fine, but using CMD does not.

### Current behavior 😯

(venv) C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0>python -m autogpt
NEWS:  Welcome to Auto-GPT!
NEWS:  Below you'll find the latest Auto-GPT News and updates regarding features!
NEWS:  If you don't wish to see this message, you can run Auto-GPT with the --skip-news flag.
NEWS:
NEWS:  Website and Documentation Site 📰📖
NEWS:  Check out https://agpt.co, the official news & updates site for Auto-GPT!
NEWS:  The documentation also has a place here, at https://docs.agpt.co
NEWS:
NEWS:  🚀 v0.3.0 Release 🚀
NEWS:  Over a week and 275 pull requests have passed since v0.2.2, and we are happy to announce
NEWS:  the release of v0.3.0! From now on, we will be focusing on major improvements rather
NEWS:  than bugfixes, as we feel stability has reached a reasonable level. Most remaining
NEWS:  issues relate to limitations in prompt generation and the memory system, which will be
NEWS:  the focus of our efforts for the next release.
NEWS:
NEWS:  Highlights and notable changes in this release:
NEWS:
NEWS:  Plugin support 🔌
NEWS:  Auto-GPT now has support for plugins! With plugins, you can extend Auto-GPT's abilities,
NEWS:  adding support for third-party services and more.
NEWS:  See https://github.com/Significant-Gravitas/Auto-GPT-Plugins for instructions and available plugins.
NEWS:
NEWS:  Changes to Docker configuration 🐋
NEWS:  The workdir has been changed from /home/appuser to /app.
NEWS:  Be sure to update any volume mounts accordingly!
NEWS:
NEWS:  ⚠️ Command `send_tweet` is DEPRECATED, and will be removed in v0.4.0 ⚠️
NEWS:  Twitter functionality (and more) is now covered by plugins, see [Plugin support 🔌]
NEWS:
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\autogpt\__main__.py", line 5, in <module>
    autogpt.cli.main()
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\venv\Lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\venv\Lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\venv\Lib\site-packages\click\core.py", line 1635, in invoke
    rv = super().invoke(ctx)
         ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\venv\Lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\venv\Lib\site-packages\click\core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\venv\Lib\site-packages\click\decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\autogpt\cli.py", line 90, in main
    run_auto_gpt(
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\autogpt\main.py", line 107, in run_auto_gpt
    workspace_directory = Workspace.make_workspace(workspace_directory)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\autogpt\workspace\workspace.py", line 52, in make_workspace
    workspace_directory = cls._sanitize_path(workspace_directory)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0\autogpt\workspace\workspace.py", line 112, in _sanitize_path
    raise ValueError("embedded null byte")
ValueError: embedded null byte

(venv) C:\Users\zacha\Dropbox (Personal)\0. Apex Haus\Coding\autogpt\Auto-GPT-0.3.0>

### Expected behavior 🤔

Should launch assistant and ask for user input.

Running with VS Code + Containers works fine. Just won't load using Windows CMD.

### Your prompt 📝

```yaml
# Paste your prompt here
```

### Your Logs 📒

```log
<insert your logs here>
```
2023-05-11 08:43:43,533 INFO NEWS: Welcome to Auto-GPT!
2023-05-11 08:43:43,535 INFO NEWS: Below you'll find the latest Auto-GPT News and updates regarding features!
2023-05-11 08:43:43,535 INFO NEWS: If you don't wish to see this message, you can run Auto-GPT with the --skip-news flag.
2023-05-11 08:43:43,535 INFO NEWS: 
2023-05-11 08:43:43,535 INFO NEWS: Website and Documentation Site 📰📖
2023-05-11 08:43:43,535 INFO NEWS: Check out https://agpt.co, the official news & updates site for Auto-GPT!
2023-05-11 08:43:43,536 INFO NEWS: The documentation also has a place here, at https://docs.agpt.co
2023-05-11 08:43:43,536 INFO NEWS: 
2023-05-11 08:43:43,536 INFO NEWS: 🚀 v0.3.0 Release 🚀
2023-05-11 08:43:43,536 INFO NEWS: Over a week and 275 pull requests have passed since v0.2.2, and we are happy to announce
2023-05-11 08:43:43,536 INFO NEWS: the release of v0.3.0! From now on, we will be focusing on major improvements rather
2023-05-11 08:43:43,536 INFO NEWS: than bugfixes, as we feel stability has reached a reasonable level. Most remaining
2023-05-11 08:43:43,537 INFO NEWS: issues relate to limitations in prompt generation and the memory system, which will be
2023-05-11 08:43:43,537 INFO NEWS: the focus of our efforts for the next release.
2023-05-11 08:43:43,537 INFO NEWS: 
2023-05-11 08:43:43,537 INFO NEWS: Highlights and notable changes in this release:
2023-05-11 08:43:43,537 INFO NEWS: 
2023-05-11 08:43:43,537 INFO NEWS: Plugin support 🔌
2023-05-11 08:43:43,537 INFO NEWS: Auto-GPT now has support for plugins! With plugins, you can extend Auto-GPT's abilities,
2023-05-11 08:43:43,537 INFO NEWS: adding support for third-party services and more.
2023-05-11 08:43:43,538 INFO NEWS: See https://github.com/Significant-Gravitas/Auto-GPT-Plugins for instructions and available plugins.
2023-05-11 08:43:43,538 INFO NEWS: 
2023-05-11 08:43:43,538 INFO NEWS: Changes to Docker configuration 🐋
2023-05-11 08:43:43,538 INFO NEWS: The workdir has been changed from /home/appuser to /app.
2023-05-11 08:43:43,538 INFO NEWS: Be sure to update any volume mounts accordingly!
2023-05-11 08:43:43,538 INFO NEWS: 
2023-05-11 08:43:43,538 INFO NEWS: ⚠️ Command `send_tweet` is DEPRECATED, and will be removed in v0.4.0 ⚠️
2023-05-11 08:43:43,538 INFO NEWS: Twitter functionality (and more) is now covered by plugins, see [Plugin support 🔌]
2023-05-11 08:43:43,538 INFO NEWS:

**Comment by ZylerKade** [comment]
Found related issue #2551 after posting this, but I do not see a fix given for 2551

**Comment by Llnski** [comment]
I am also having this issue, running in fresh Windows 10 VM.

**Comment by ZylerKade** [comment]
@Llnski I was never able to get this working using Windows CMD, however I ended up installing VC Code and running it with Dev Containers, was able to get it running that way.

**Comment by Llnski** [comment]
@ZylerKade ; I was finally able to get it working in Powershell(although I'd assume this fix would be just as good in cmd) by installing [https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170](https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170).

**Comment by JungWonYong** [comment]
Supplementally, it seems... using  virtual environment of python may cause problem on other OS. 

I am using macos, autogpt do infinite loop for installing python module when situation of AutoGPT generate python script for subtask.
ex) Error: no xxx module → install xxx module with pip → installed successfully → Error: no xxx module → ..... same thing loop infinitely.

I think the module installed to virtual environment (It uses path when installing module, when running pip.), but use system python to execute .pl script what is AutoGTP generate to do subtask. 
(Of course I do unset PATH to system python and I checked not displayed with "which" command.)

To avoid this problem, I did install 3.11 with package manager like homebrew, and set PATH to it. 
And then using it without venv.

Checked environment:
------------------------------
hardware : M1 macbook air
OS : Macos 13.3.1 (a)
Python : 3..11 with homebrew
AutoGPT : stable branch head (0.3.1)
